server/src/server.ts

```diff
@@ -295,5 +314,214 @@ function getDiagnostic(info: any, map: any): Diagnostic {
+                return makeLocation(pathData.jsonMap.pointers['/shaders/' + result]);
+            }
+        }
+        else {
+            if (part === 'nodes' || part === 'children' || part === 'joints') {
+                inNodes = true;
+            }
+            else if (part === 'channels') {
+                inChannels = true;
+                currentAnimationPath = currentPath.substring(0, currentPath.length - '/channels'.length);
+            }
+            else if (result.uri !== undefined) {
+                if (!result.uri.startsWith('data:') && !currentPath.startsWith('/images/')) {
+                    return makeLocation(null, Url.resolve(textDocumentPosition.textDocument.uri, result.uri));
+                } else {
+                    let uri = 'gltf-dataUri://' + encodeURIComponent(Uri.parse(textDocumentPosition.textDocument.uri).fsPath) + currentPath;
+                    return makeLocation(null, uri);
+                }
+            } else if (part === 'accessors') {
+                inAccessors = true;
+            } else if (inAccessors && !path.includes('bufferView')) {
+                let uri = 'gltf-dataUri://' + encodeURIComponent(Uri.parse(textDocumentPosition.textDocument.uri).fsPath) + currentPath;
+                return makeLocation(null, uri);
+            }
+        }
+    }
+
+    return null
+});
+
+connection.onHover((textDocumentPosition: TextDocumentPositionParams): Hover => {
+    let pathData = getPath(textDocumentPosition);
+    if (!pathData) {
+        return null;
+    }
+
+    let path = pathData.path;
+
+    if (path.startsWith('/images/')) {
+        let imageData = getFromPath(pathData.jsonMap.data, path);
+        if (imageData) {
+            if (!path.endsWith('/uri')) {
+                imageData = imageData.uri;
+            }
+            if (!imageData.startsWith('data:')) {
+                imageData = Url.resolve(textDocumentPosition.textDocument.uri, imageData);
+            }
+            let contents: MarkedString[] = [`![${path}](${imageData})`];
+            return {
+                contents: contents,
+                range: Range.create(pathData.start, pathData.end)
+            };
+        }
+    }
+
+    if (path.includes('diffuseFactor') || path.includes('specularFactor') || path.includes('baseColorFactor'))
+    {
+        if (!Number.isNaN(parseInt(path.charAt(path.length-1)))) {
+            path = path.substring(0, path.length-2);
+        }
+        let colorData = getFromPath(pathData.jsonMap.data, path);
+        let red = Math.round(colorData[0] * 255);
+        let blue = Math.round(colorData[1] * 255);
+        let green = Math.round(colorData[2] * 255);
+        let hexColor = ((red < 16) ? '0' : '') + red.toString(16) + ((blue < 16) ? '0' : '') +  blue.toString(16) + ((green < 16) ? '0' : '') +  green.toString(16);
+        let contents: MarkedString[] = [`![${hexColor}](https://placehold.it/50/${hexColor}/000000?text=+)`];
```

Comment:
I do try to keep this extension free of references to the Internet, so it can work offline.  But we'll let this one slide since it's just a color swatch and the offline folks can live without it.

Reply:
If we could use Html directly instead of Markup then we'd have some options.